server/app/mutations/stacks/create.rb

```diff
@@ -45,11 +51,28 @@ def validate_services
       add_error("services.#{error.service}.links", :recursive, error.message)
     end
 
+    def validate_parent
+      _, parent_name = if self.inputs.has_key?(:parent)
+        self.inputs.delete(:parent)[:id].split('/')
+      else
+        [self.grid.name, self.inputs.delete(:parent_name)]
```

Comment:
:-1: for mutating `self.inputs` in the validate methods

Reply:
Yes, fixed.